pegaprox/core/config_vault.py

```
@@ -0,0 +1,1135 @@
+        error[:500], object_key, object_key, _now(), provider_id,
+    ))
+    db.conn.commit()
+
+
+def _retention_cleanup(adapter, prefix: str, keep: int, current_key: str):
+    try:
+        objects = adapter.list(prefix)
+        keys = [item['key'] for item in objects if item.get('key') != current_key]
+        for key in keys[max(0, keep - 1):]:
+            adapter.delete(key)
+    except Exception as exc:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)